frontend/src/api/index.ts

```diff
@@ -10,13 +11,36 @@ let api: Client | undefined;
 let waiters: Array<Waiter> = [];
 
 export default {
-  init: () => {
+  init: async (actions) => {
     api = new Client(METABACKEND_URL, METABACKEND_NETWORK);
-    api.initialize().then(()=>{
-        // Wake up sleepers waiting for API.
-        for (let i = 0; i < waiters.length; i++) {
-          waiters[i](api);
+    (async () => {
```

Comment:
`init` is not async, it returns immediately since you're callng a nested asynced-promise inside without returning it or waiting for results.
Suggestion for cleaner code -> remove the nested async function, do everything in the async init, remove promises and all thens, and just use await and try/catch